experimental/language-generation/docs/Functions-injected-from-LG.md

```diff
@@ -0,0 +1,222 @@
+||||
+
+*Example*
+
+This example converts an collection of objects to an activityAttachment.
+Suppose we want to convert a JSON file to herocard attachment. 
+the content in the herocard.json: 
+
+```
+{
+  "title": "titleContent",
+  "text": "textContent",
+  "Buttons": [
+    {
+      "type": "imBack",
+      "Title": "titleContent",
+      "Value": "textContent",
+      "Text": "textContent"
+    }
+  ],
+  "tap": {
+    "type": "@{type}",
+    "title": "@{title}",
+    "text": "@{title}",
+    "value": "@{value}"
+  }
+}
+```
+
+By calling ActivityAttachment in a template, type, title, value are passed from template name.
+
+```
+# externalHeroCardActivity(type, title, value)
+[Activity
+    attachments = @{ActivityAttachment(json(fromFile('.\\herocard.json')), 'herocard')}
+]
+```
+
+And returns a herocard:
+
+```
+{
+    "lgType" = "attachment",
+    "contenttype" = "herocard",
+    "content" = {
+        "title": "titleContent",
+        "text": "textContent",
+        "Buttons": [
+            {
+            "type": "imBack",
+            "Title": "titleContent",
+            "Value": "textContent",
+            "Text": "textContent"
+            }
+        ],
+        "tap": {
+            "type": "@{type}",
+            "title": "@{title}",
+            "text": "@{title}",
+            "value": "@{value}"
+        }
+    }
+}
+```
+
+<a name="template"></a>
+### template
+
+Return the evaluated result of the template name and parameters.
+
+```
+template(<collection-of-objects>)
+```
+
+| Parameter | Required | Type | Description |
+| --------- | -------- | ---- | ----------- |
+| <*collection-of-objects*> | Yes | Array or object  | A collection of objects|
+|||||
+
+| Return value | Type | Description |
+| ------------ | -----| ----------- |
+| <*evaluated-result*> | Object | the result evaluated from the template as a function  |
+||||
+
+*Example*
+
+This example evaluates the result of calling the template as a function :
+Suppose we have template:
+    
+    # welcome(userName)
+
+    - Hi @{userName}
+
+    - Hello @{userName}
+
+    - Hey @{userName}
+
+```
+template("welcome", new { userName = "DL" }.ToString())
+```
+
+And returns one of the results:
+
+```
+Hi DL
+Hello DL
+Hey DL
+```
+
+<a name="fromFile"></a>
+
+### fromFile
+
+Return the evaluated result from the expression in the given file
+
+```
+fromFile(<filepath>)
+```
+
+| Parameter | Required | Type | Description |
+| --------- | -------- | ---- | ----------- |
+| <*filepath*> | Yes | string  | path of the a file |
+|||||
+
+| Return value | Type | Description |
+| ------------ | -----| ----------- |
+| <*function*> | Function | A function convert object to activityAttachment |
+||||
+
+*Example*
+
+This example evaluate the result from the given filepath :
+Suppose we have a file whose filepath is:  
+
+`/home/user/test.txt`.
+
+The content of the file is 
+
+`add(1,2)`
+
+```
+fromFile('/home/user/test.txt')
```

Comment:
this is not right. `fromFile` just return the string content of the resource that with the absolute path or the relative path relative th the current LG file.